hell/adventcalendar/2025/35/index.md

```diff
@@ -1,35 +1,102 @@
 
+## bgsound
 
-<p class="highlight"><strong>Note:</strong> Some text.</p>
+`<bgsound>` was a way to play sound in the background, because everyone loves web pages making sounds they didn't ask for at them. It was not standardised, and exclusively part of Internet Explorer. The `<audio>` tag is the much more courteous modern equivalent.
 
-```html
-<h1> 
-  <a href="/">
-    Hello World
-  </a>
-</h1>
-```
+I find it endlessly humorous that the old HTML Wiki's [example section for the tag](https://www.w3.org/html/wiki/Elements/bgsound#Examples) simply states, 'No, really. Don't use it.'. Many of the old and obscure elements share the same detail.
+
+## Framesets
+
+In the olden days, people used 'frames'. Not iframes mind you, just frames. You can think of their usage vaguely like the HTML imports we have now... Wait, what do you mean we still don't have HTML imports?
+
+I digress. This was long before single-page applications (SPAs) or anything of that nature, but were widely used as a way to aid navigation without needing to reload the page. You could have a navigation frame that persisted while the content frame changed.
+
+`<frame>` is more or less the same as our modern `<iframe>`, though we use iframes much more for embedding than page layout. `<frameset>` holds these frames, so you could write something like `<frameset cols="50%, 50%">` to have two frames next to each other.
+
+The frameset approach had quite a few problems. Accessibility was really poor, and because everything was a single page, it could be hard to link to a specific section. ~~Thankfully we've come a long way since then, and developers take great care with accessibility and ensuring their SPAs respect the web platform.~~
+
+Similarly to how `<noscript>` is triggered if JavaScript isn't supported, `<noframes>` allows you to set content if frames aren't supported.
+
+While they absolutely shouldn't be used anymore, unlike much of what this article covers, frames and framesets are still supported. They may have been deprecated in HTML5, but the web's backward compatibility keeps them going.
+
+## Code
+
+`<pre>` and `<code>` aren't strangers to us. We're good friends, no matter the horrific scripts we may force them to hold. However, pre-`<pre>` and `<code>` there were plenty of other implementations in the wind.
+
+The e**x**a**mp**le element, `<xmp>`, is like our modern `<pre>`, but doesn't interpret special characters like `<`, `>`, or `&`. With `<pre>`, we often need to escape these values to avoid Unwanted Shenanigans™.
+
+`<listing>` is yet another step in presenting code on a page. It was intended as a way to display text on a page as typed but never got proper support and became obsolete with HTML5. Modern browsers handle it the same as `<pre>`.
```

Comment:
does “as typed” mean it preserved white space etc? any ideas on how to make that more obvious?